src/lib/webbrowser.js

```diff
@@ -0,0 +1,38 @@
+var $builtinmodule = function(name){
+    var mod = {};
+    var inBrowser = (typeof window != "undefined") && (typeof window.navigator != "undefined");
+
+    function open_tab(url) {
+        Sk.builtin.pyCheckType("url", "string", Sk.builtin.checkString(url));
+        if (!inBrowser) {
+            return Sk.builtin.bool.false$;
+        }
+
+        url = url.$jsstr();
+        window.open(url, "_blank");
+
+        return Sk.builtin.bool.true$;
+    }
+
+    mod.open = new Sk.builtin.func(function open(url) {
+        Sk.builtin.pyCheckArgsLen("open", arguments.length + 1, 1, 3);
+        return open_tab(url);
+    });
+
+    mod.open_new = new Sk.builtin.func(function open_new(url) {
+        Sk.builtin.pyCheckArgsLen("open_new", arguments.length + 1, 1, 1);
+        return open_tab(url);
+    });
+
+    mod.open_new_tab = new Sk.builtin.func(function open_new_tab(url) {
+        Sk.builtin.pyCheckArgsLen("open_new_tab", arguments.length + 1, 1, 1);
+        return open_tab(url);
+    });
+
+    mod.get = new Sk.builtin.func(function get() {
+        Sk.builtin.pyCheckArgsLen("get", arguments.length, 0, 1);
+        return $builtinmodule;
```

Comment:
It seems that `webbrowser.get(using=None)` returns a browser object (seems they support multiple browsers?) which also have `open/open_new/open_new_tab` as methods.
However we only have one choice for browser JS.. so I'd like to return the module it self, am I doing correct by `return $builtinmodule;`?

Reply:
I don't think we need to implement this. 

If you want to then I would probably create a class object with the same methods as the module
here's the BaseBrowser class from Cpython
https://github.com/python/cpython/blob/c75330605d4795850ec74fdc4d69aa5d92f76c00/Lib/webbrowser.py#L139

For a simple class object written in JS I'd checkout the `re.js` file which has a couple.

Reply:
Done, thank for the instructions!

BTW, I also noticed that many functions in `re.js` are not following the rules we're now using in `webbrowser.js`:
1. `function some_name(...)`
2. `Sk.builtin.pyCheckArgsLen`
2. `Sk.builtin.pyCheckType`

Which may result in confusing error message when users using them wrongly. Is this intentional for simplicity?

Reply:
Yes - there are two approaches to define a function. 

The approach in this module or the approach in re.js - that approach might do the equivalent of

```javascript
function open_new(url) {
  // only check type 
}
open_new.co_varnames = ["url"]
mod.open_new = new Sk.builtin.func(open_new)
```

the above tells the compiler that there is only one are called url. 

It has the benefit that the end user can do `url='foo'` as they can in python. But it's not the most intuitive interface.

It gets more complex the more complicated your call signature is.